plugins/microservices/src/test_msp_functions.cpp

```diff
@@ -0,0 +1,71 @@
+/// \file
+
+#include "irods/irods_ms_plugin.hpp"
+#include "irods/msParam.h"
+
+#include "msi_assertion_macros.hpp"
+
+namespace
+{
+    auto msi_impl(RuleExecInfo* _rei) -> int
+    {
+        IRODS_MSI_TEST_BEGIN("testing msp_to_ function: msp_to_string")
+        {
+            auto ret = msp_to_string(nullptr, nullptr);
+            IRODS_MSI_ASSERT(ret == SYS_NULL_INPUT);
+        }
+        {
+            char* good = nullptr;
+            auto ret = msp_to_string(nullptr, &good);
+            IRODS_MSI_ASSERT((ret == SYS_NULL_INPUT && good == nullptr));
```

Comment:
If `ret` is `const`, this is fine. If it's not `const`, consider swapping the order of the operands. That protects us from `ret = SYS_NULL_INPUT`.